src/github.com/getlantern/flashlight/proxiedsites/proxiedsites.go

```diff
@@ -0,0 +1,70 @@
+package proxiedsites
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sync"
+
+	"github.com/getlantern/golog"
+	"github.com/getlantern/proxiedsites"
+
+	"github.com/getlantern/flashlight/config"
+	"github.com/getlantern/flashlight/ui"
+)
+
+var (
+	log = golog.LoggerFor("proxiedsites-flashlight")
+
+	uichannel  *ui.UIChannel
+	startMutex sync.Mutex
+)
+
+func Configure(cfg *proxiedsites.Config) {
+	delta := proxiedsites.Configure(cfg)
+	startMutex.Lock()
+	if uichannel == nil {
+		start()
+	} else if delta != nil {
+		b, err := json.Marshal(delta)
+		if err != nil {
+			log.Errorf("Unable to publish delta to UI: %v", err)
+		}
+		uichannel.Out <- b
+	}
+	startMutex.Unlock()
+}
+
+func start() {
+	// Register the PAC handler
+	url := ui.Handle("/proxy_on.pac", http.HandlerFunc(proxiedsites.ServePAC))
+	log.Debugf("Serving PAC file at %v", url)
+
+	// Establish a channel to the UI for sending and receiving updates
+	uichannel = ui.NewChannel("/data", func(write func([]byte) error) error {
```

Comment:
@atavism @xiam UIChannel is an abstraction that I added to represent a channel of communication to the UI.  All browser windows are represented by this channel, so that when any of them write, we get the message, and if we write to the channel, all of the browsers get the message.  I use this abstraction here for the proxiedsites functionality, and @xiam can use it elsewhere for the geoip functionality (just use a different path than `/data`).